src/main/java/net/rakugakibox/springbootext/logback/access/tomcat/RequestAttributeRegardingAccessEvent.java

```diff
@@ -0,0 +1,91 @@
+package net.rakugakibox.springbootext.logback.access.tomcat;
+
+import ch.qos.logback.access.spi.AccessEvent;
+import ch.qos.logback.access.spi.ServerAdapter;
+import org.apache.catalina.AccessLog;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * AccessEvent that regards requestAttributes that may be set by other valves.
+ */
+public class RequestAttributeRegardingAccessEvent extends AccessEvent {
+
+    private String remoteAddress;
+    private String remoteHost;
+    private String protocol;
+    private Integer localPort;
+
+    public RequestAttributeRegardingAccessEvent(HttpServletRequest httpRequest, HttpServletResponse httpResponse,
+                                                ServerAdapter adapter) {
+        super(httpRequest, httpResponse, adapter);
+        prepareForDeferredProcessing();
+    }
+
+    @Override
+    public String getRemoteAddr() {
+        if (remoteAddress == null) {
+            HttpServletRequest request = getRequest();
+            if (request != null) {
+                remoteAddress = (String) request.getAttribute(AccessLog.REMOTE_ADDR_ATTRIBUTE);
+                if (remoteAddress == null) {
+                    remoteAddress = request.getRemoteAddr();
+                }
+            } else {
+                remoteAddress = NA;
```

Comment:
A static variable that indicates that the value is not present (coming from the IAccessEvent interface).